drivers/display/display_nrf_led_matrix.c

```diff
@@ -340,7 +354,7 @@ static void prepare_pixel_pulse(const struct device *dev,
 
 	/* First timer channel is used for timing the period of pulses. */
 	nrf_timer_cc_set(dev_config->timer, 1 + channel_idx, pulse);
-	NRF_GPIOTE->CONFIG[dev_data->gpiote_ch[channel_idx]] = gpiote_cfg;
+	gpiote[channel_idx].p_reg->CONFIG[dev_data->gpiote_ch[channel_idx]] = gpiote_cfg;
```

Comment:
This cannot be done this way for the general case. GPIOTE channels here may be used with various column pins, not necessarily with the same ones that were used to fill the `gpiote` array at particular indexes. See https://github.com/zephyrproject-rtos/zephyr/blob/5a2b29169780ec9cc69865f23bed6e1ca76bea07/drivers/display/display_nrf_led_matrix.c#L334-L335
I think we should rather limit the driver to use one GPIOTE instance and ensure at build time that all column GPIOs can use that instance. I'll dig into this further and propose something specific.